### PR TITLE
Add missing index.ts

### PR DESCRIPTION
**Pull request by philippjfr**
Checklist
* [x] Used a [fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by philippjfr** [comment]
So I don't think this worked yet, it's still saying:

>  Compiling TypeScript (27 files)

when it should be 28 files.

**Comment by philippjfr** [comment]
Despite the patch it still says:

>    warning: no files found matching 'panel/index.ts'

**Comment by philippjfr** [comment]
Might be this:

>Patch level ambiguous, selecting least deep

Just not clear what's ambiguous about the patch.

File: recipe/add-index-ts.patch [hunk 1]
```diff
@@ -0,0 +1,11 @@
+diff --git a/panel/index.ts b/panel/index.ts
+new file mode 100644
+index 00000000..98b8b880
+--- /dev/null
++++ b/panel/index.ts
```
Comment (philippjfr) [inline recipe/add-index-ts.patch:5]:
```suggestion
+++ b/panel/panel/index.ts
```

**Comment by philippjfr** [comment]
This does not actually fix the issue yet, the patch adding the panel/index.ts does not seem to be working as it can't be found during the build.

**Comment by ocefpaf** [comment]
> This does not actually fix the issue yet, the patch adding the panel/index.ts does not seem to be working as it can't be found during the build.

Looking into that...

**Comment by philippjfr** [comment]
Here's what it says about the patch, but I can't make sense of it.

```
Applying patch: '/home/conda/recipe_root/add-index-ts.patch'
Patch level ambiguous, selecting least deep
Trying to apply patch as-is
INFO:conda_build.source:Trying to apply patch as-is
INFO conda_build.source:apply_patch(598): Trying to apply patch as-is
patching file b/panel/index.ts
```

**Comment by ocefpaf** [comment]
Adding new files via patch is a bit awkward. Let's try copying that file from the recipe director into the source directory.

**Comment by philippjfr** [comment]
Thanks, that should work. If this is not generally supported maybe we should notify the 20+ other feedstocks that also add files via patch? Maybe it's working for them though.

**Comment by ocefpaf** [comment]
> Thanks, that should work. If this is not generally supported maybe we should notify the 20+ other feedstocks that also add files via patch? Maybe it's working for them though.

It is supported to add via patch. It is just a but less awkward copying instead. And a bit clearer.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -7,6 +7,7 @@ package:
 source:
   url: https://pypi.io/packages/source/p/panel/panel-{{ version }}.tar.gz
   sha256: 53340615f30f67f3182793695ebe52bf25e7bbb0751aba6f29763244350d0f42
+  patches:
     - index-ts-fix.patch
```
Comment (ocefpaf) [inline recipe/meta.yaml:11]: Yep. We need that one!

**Comment by philippjfr** [comment]
Hurray:

```
  Compiling TypeScript (28 files)
  ...
  copying panel/index.ts -> build/lib/panel
```

Thanks again for your help!

**Pull request merged**